### PR TITLE
refactor: remove all unnecessary @ts-ignore annotations

### PR DESCRIPTION
**Pull request by michaelfig**
Either updates to the Typescript version we use in agoric-sdk, or cargo-culting means many of our `@ts-ignore` annotations are unnecessary.

I suggest that we should avoid `@ts-ignore` whenever we can: only use it when Typescript really has a bug we can't work around, such as https://github.com/Agoric/agoric-sdk/pull/2438#discussion_r576948615.  Then, link to the corresponding open issue on the TypeScript repository in the same line as `@ts-ignore`.

### CODE REVIEW

File: packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js [hunk 1]
```diff
@@ -12,7 +12,6 @@ const decoder = new TextDecoder();
 
 // eslint-disable-next-line no-unused-vars
 function workerLog(first, ...args) {
-  // @ts-ignore
```
Comment (dckc) [inline packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js:15]: I guess this dates back before the `console.log` below was commented out.

Here's hoping for a logging configuration API that doesn't involve editing source code :)

File: packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js [hunk 2]
```diff
@@ -194,23 +193,26 @@ function makeWorker(port) {
...
-      virtualObjectCacheSize,
+      cacheSize,
       // TODO: lsgc? API drift?
     );
 
     const endowments = {
       ...ls.vatGlobals,
       console: makeConsole(`SwingSet:vatWorker`),
       assert,
-      // @ts-ignore bootstrap provides HandledPromise
```
Comment (dckc) [inline packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js:242]: I don't understand why this is no longer necessary. Help?

File: packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js [hunk 3]
```diff
@@ -251,8 +253,7 @@ function makeWorker(port) {
   });
 }
 
-// @ts-ignore xsnap provides issueCommand global
```
Comment (dckc) [inline packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js:285]: again, why is this no longer necessary? I suppose I should check change comments...

**Comment by katelynsills** [comment]
The `@ts-ignore` annotations aren't cargo-culting. I put them there on purpose. In my local settings, I have type checking on by default (even without `ts-check`) and it is helpful to me to be able to type check the tests, and where the types are intentionally wrong in the tests (normally a t.throws test), to suppress that error. As it's not hurting anyone, and it's my preference, I would like to leave these as they are.

File: packages/marshal/src/marshal.js [hunk 4]
```diff
@@ -300,6 +300,15 @@ const makeRemotableProto = (oldProto, allegedName) => {
   });
 };
 
+/**
+ * Throw if val is not the correct shape for the prototype of a Remotable.
+ *
+ * TODO: It would be nice to typedef this shape and then declare that this
+ * function asserts it, but we can't declare a type with PASS_STYLE from JSDoc.
+ *
+ * @param {{ [PASS_STYLE]: string, [Symbol.toStringTag]: string, toString: () =>
```
Comment (dckc) [inline packages/marshal/src/marshal.js:352]: odd use of whitespace. I'd rather see each member on its own line, for consistency with prettify

File: packages/xsnap/lib/lockdown-shim.js [hunk 5]
```diff
@@ -1,4 +1,3 @@
-// @ts-ignore
```
Comment (dckc) [inline packages/xsnap/lib/lockdown-shim.js:1]: I don't understand why this is no longer necessary. hm.

**Review by katelynsills — CHANGES_REQUESTED** [review]
Please keep the `@ts-ignore` above `t.throws` and `t.throwsAsync`

**Review by dckc — COMMENTED** [review]
I'd like to understand better why some of these are no longer necessary. Not critical.

I'm not sure about the Zoe stuff. I guess I'll stand by for Kate's comments... but that doesn't seem like an algorithm that terminates.  Splitting the PR would make for a more clear process, but maybe that's more trouble than it's worth.

**Comment by michaelfig** [comment]
> The `@ts-ignore` annotations aren't cargo-culting. I put them there on purpose. In my local settings, I have type checking on by default (even without `ts-check`) and it is helpful to me to be able to type check the tests, and where the types are intentionally wrong in the tests (normally a t.throws test), to suppress that error. As it's not hurting anyone, and it's my preference, I would like to leave these as they are.

Sorry, I had assumed those files were running with `@ts-check` so when I removed the `@ts-ignore` I didn't see any effect.  
I will revert those changes.

Would you like me to add `@ts-check` to them?

**Comment by katelynsills** [comment]
> Sorry, I had assumed those files were running with `@ts-check` so when I removed the `@ts-ignore` I didn't see any effect.
> I will revert those changes.
> 
> Would you like me to add `@ts-check` to them?

That would be great, thanks!

File: packages/marshal/src/marshal.js [hunk 6]
```diff
@@ -349,8 +356,8 @@ function assertCanBeRemotable(val) {
   const keys = ownKeys(descs); // enumerable-and-not, string-or-Symbol
   keys.forEach(key => {
     assert(
-      // @ts-ignore
-      !('get' in descs[key]),
+      // Typecast needed due to https://github.com/microsoft/TypeScript/issues/1863
+      !('get' in descs[/** @type {string} */ (key)]),
```
Comment (erights) [inline packages/marshal/src/marshal.js:403]: @warner I thought I saw something from you saying that you tried this cast and it didn't work for you. Would be good to understand why vs why it works for @michaelfig . Possibly I screwed something up in relaying the cast syntax as I remembered it? Attn @Chris-Hibbert also on that thread.

Comment (michaelfig) [reply packages/marshal/src/marshal.js:403]: > Possibly I screwed something up in relaying the cast syntax as I remembered it?

Yeah, you had said `/** type{string} */` which didn't parse.

Don't worry about it, though.

**Comment by michaelfig** [comment]
@katelynsills would you be able to take a look at the documentation snippets failure at: https://github.com/Agoric/agoric-sdk/pull/2441/checks?check_run_id=1913294837#step:12:10

Thanks!

**Comment by katelynsills** [comment]
> @katelynsills would you be able to take a look at the documentation snippets failure at: https://github.com/Agoric/agoric-sdk/pull/2441/checks?check_run_id=1913294837#step:12:10
> 
> Thanks!

Yup!

File: packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js (hunk 2 above)
Comment (michaelfig) [reply packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js:242]: If you use an explicit `globalThis.foo`, typescript doesn't complain, but it does if you just use `foo`.

File: packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js (hunk 3 above)
Comment (michaelfig) [reply packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js:285]: As above.

File: packages/marshal/src/marshal.js (hunk 4 above)
Comment (michaelfig) [reply packages/marshal/src/marshal.js:352]: I'm a victim of my own auto-wrap-comments tooling.

**Review by katelynsills — APPROVED** [review]
LGTM! Thanks for doing this cleanup

**Comment by katelynsills** [comment]
Is this waiting on something? The documentation checks are passing 👍

**Comment by michaelfig** [comment]
> Is this waiting on something? The documentation checks are passing 👍

Not waiting on anything now!  Thanks for the prod.

**Pull request merged**